update the documentation and change log for Windows release

rawahars opened the pull request:
## Summary
In this PR, we are updating the documentation to account for Windows release.

This includes notes about the following in our documentation-
- Supported Windows tags
- Supported Windows releases
- SSM parameters for Windows images
- Monthly security update of Windows images
- Change log documentation

Moreover, we are also updating the 2.28.3 change log to include Windows information.

## Description of changes:
updated documentation for Windows release

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

PettitWesley commented on README.md:
```diff
@@ -80,22 +80,33 @@ If you think you’ve found a potentially sensitive security issue, please do no
-Each release updates the `latest` tag and adds a tag for the version of the image. The `stable` tag is also available which marks a release as the latest stable version. Deploying `latest` to prod without going through a test stage first is not recommended.
+Each release updates the `latest` tag and adds a tag for the version of the image. The `stable` tag is also available which marks a release as the latest stable version.
 
+For Windows images, we update the `windowsservercore-latest` tag and add a tag as `<VERSION>-windowsservercore`. The stable tag is available as `windowsservercore-stable`. We update all the supported versions each month when [Microsoft releases the latest security
```
Can you make it a header we can link to like the arm vs amd below?

rawahars commented:
We need to change the change log in this PR.

PettitWesley commented on README.md:
```diff
@@ -145,7 +160,9 @@ Parameters:
 
-You should lock your deployments to a specific version tag. We guarantee that these tags will be immutable- once they are released the will not change.
+You should lock your deployments to a specific version tag. We guarantee that these tags will be immutable- once they are released the will not change. 
+Windows images would be updated each month to include the latest security patches in the base layers but the contents of the image would not change in a tag. 
```
nit: would be => will be

PettitWesley commented on README.md:
````diff
@@ -185,6 +202,17 @@ The official way to find the ECR image URIs for your region is to use the SSM Pa
 aws ssm get-parameters-by-path --path /aws/service/aws-for-fluent-bit/
 ```
 
+#### Networking issue with Windows containers when using `async` DNS resolution by plugins
````
I think we need this but it does not fit here under the image distribution section. 

Instead, I would prefer this be added as a section in the debugging.md and then link it under the main tag in the readme for windows support.

PettitWesley replied:
Let's create a windows specific section of the debugging guide and then link to it and specifically call out that the guidance must be followed if you run windows containers in default namespace

PettitWesley replied:
I mean in the readme say "Check out the Windows debugging guide. There is a known issue with..." something like that

PettitWesley replied:
windows section of the debugging guide

PettitWesley commented on README.md:
````diff
@@ -185,6 +202,17 @@ The official way to find the ECR image URIs for your region is to use the SSM Pa
+```
+net.dns.mode LEGACY
+```
+This workaround is applicable only for Fluent Bit 1.9 as of now.
````
Sorry wait- do we know that its actually fixed in 2.0?

Until its fixed, we should probably not restrict this guidance to a version, since then it will become out of date

rawahars replied:
Yes right. We have not validated the same in 2.0 till now. 

Let me remove this statement for now.

PettitWesley commented on README.md:
```diff
@@ -198,6 +216,15 @@ You can use the AWS Fluent Bit plugins with [td-agent-bit](https://docs.fluentbi
+### Running `aws-for-fluent-bit` Windows containers
+You can run `aws-for-fluent-bit` Windows containers using the image tags as specified [here](#windows-images). These are distributed as multi-arch images with the manifests for the supported Windows releases as specified above.
+
+For more details about running Fluent Bit Windows containers in Amazon EKS, please visit our blog [here](https://aws.amazon.com/blogs/containers/centralized-logging-for-windows-containers-on-amazon-eks-using-fluent-bit/).
```
IIRC this matches amazon blog writing style guide

rawahars commented:
Rebased the PR with latest mainline and added the following changes in the latest revision-
- Added a section about enabling core dumps for Windows images
- Added the changelog for first Windows release

PettitWesley commented on windows.versions:
```diff
@@ -1,15 +1,16 @@
-      "flexBison": "2.5.22"
+      "flexBison": "2.5.22",
+      "latest": true,
+      "stable": true
```
I think the first release should have stable: false

rawahars replied:
This is required by `sync` task which needs the stable version tag.

We can maybe add in change log that this is marked as stable but not been properly vetted yet.

rawahars commented on CHANGELOG.md:
```diff
@@ -2,13 +2,22 @@
 ### 2.28.4
 This release includes:
+
+For Linux images:
+* Fluent Bit [1.9.9](https://fluentbit.io/announcements/v1.9.9/)
+* Amazon CloudWatch Logs for Fluent Bit 1.9.1
+* Amazon Kinesis Streams for Fluent Bit 1.10.1
+* Amazon Kinesis Firehose for Fluent Bit 1.7.1
+
+For Windows images:
```
Maybe for future if things diverge. Should we keep it same for now?
